spopt/locate/coverage.py

```diff
@@ -371,59 +461,90 @@ def from_cost_matrix(
-        ...                            columns=["geometry"])
+        >>> streets = spaghetti.element_as_gdf(ntw, arcs=True)
+        >>> streets_buffered = geopandas.GeoDataFrame(
+        ...     geopandas.GeoSeries(streets["geometry"].buffer(0.2).unary_union),
+        ...     crs=streets.crs,
+        ...     columns=["geometry"]
+        ... )
 
         Simulate points belong to lattice
 
-        >>> demand_points = simulated_geo_points(street_buffered, needed=100, seed=5)
-        >>> facility_points = simulated_geo_points(street_buffered, needed=5, seed=6)
+        >>> demand_points = simulated_geo_points(streets_buffered, needed=100, seed=5)
+        >>> facility_points = simulated_geo_points(streets_buffered, needed=5, seed=6)
 
         Snap points to the network
 
         >>> ntw.snapobservations(demand_points, "clients", attribute=True)
-        >>> clients_snapped = spaghetti.element_as_gdf(ntw, pp_name="clients", snapped=True)
+        >>> clients_snapped = spaghetti.element_as_gdf(
+        ...     ntw, pp_name="clients", snapped=True
+        ... )
         >>> ntw.snapobservations(facility_points, "facilities", attribute=True)
-        >>> facilities_snapped = spaghetti.element_as_gdf(ntw, pp_name="facilities", snapped=True)
+        >>> facilities_snapped = spaghetti.element_as_gdf(
+        ...     ntw, pp_name="facilities", snapped=True
+        ... )
 
         Calculate the cost matrix
         >>> cost_matrix = ntw.allneighbordistances(
-        ...    sourcepattern=ntw.pointpatterns["clients"],
-        ...    destpattern=ntw.pointpatterns["facilities"])
+        ...     sourcepattern=ntw.pointpatterns["clients"],
+        ...     destpattern=ntw.pointpatterns["facilities"]
+        ... )
 
         Create LSCPB instance from cost matrix
 
-        >>> lscpb_from_cost_matrix = LSCPB.from_cost_matrix(cost_matrix, max_coverage=8, pulp.PULP_CBC_CMD(msg=False)
-        >>> lscpb_from_cost_matrix = lscpb_from_cost_matrix.solve(pulp.PULP_CBC_CMD(msg=False))
+        >>> lscpb_from_cost_matrix = LSCPB.from_cost_matrix(
+        ...     cost_matrix, service_radius=8, solver=pulp.PULP_CBC_CMD(msg=False)
+        ... )
+        >>> lscpb_from_cost_matrix = lscpb_from_cost_matrix.solve()
 
         Get facility lookup demand coverage array
 
-        >>> lscpb_from_cost_matrix.fac2cli
+        >>> for fac, cli in enumerate(lscpb_from_cost_matrix.fac2cli):
+        ...     print(f"facility {fac} serving {len(cli)} clients")
+        facility 0 serving 0 clients
+        facility 1 serving 63 clients
+        facility 2 serving 85 clients
+        facility 3 serving 92 clients
+        facility 4 serving 0 clients
+
+        Get the percentage of clients covered by more than one facility.
+
+        >>> round(lscpb_from_cost_matrix.backup_perc, 3)
+        88.0
+
+        88% of clients are covered by more than 1 facility
 
         """
         if predefined_facilities_arr is not None:
```

Comment:
Should predefined facilities be equal for LSCP and LSCPB when solving LSCPB? Does that make sense give an option to pass another column to distinguish predefined sites between these models?

Reply:
This is an interesting question – I think they should be equal. Since the "predefined"ness of the LSCPB is derived from the LSCP solution, those facilities that are required to be in the LSCPB solution must also be passed into the LSCP.

Can you think of a scenario where this would not be the case? I am open to different ideas/opinions.

Reply:
@qszhao & @ljwolf, how do you feel about this point that @gegen07 brought up here?

Reply:
I couldn't think of a case that this occurs. Hmm, but I imagined that if we solve the LSCP with predefined facilities we will already choose them in the results, so these sites will not interfere with LSCPB solving. So, we can disconsider the predefined locations for LSCPB, if we assume they are the same. What do you think @jGaboardi?

Reply:
But we still need to pass the predefined facilities into the LSCPB instance in order to ensure they are part of the LSCPB solution. This is because the standard LSCPB takes into consideration the objective of the LSCP (*number* of facilities, but not *which* facilities). So I think the correct answer is to leave it as is.

Reply:
OK! I think we can go on and merge this PR.